scripts/globals/pets/avatar.lua

```diff
@@ -0,0 +1,288 @@
+            local partyHP = false
+            local party = master:getParty()            
+            for _,member in ipairs(party) do
+                if member:getHPP() < 50 then                    
+                    partyHP = true
+                    break
+                end
+            end      
+            if masterHP1 and partyHP and level > 15 then --curaga                
+                pet:castSpell(level >= 91 and 11 or level >= 71 and 10 or level >= 51 and 9 or level >= 31 and 8 or 7, master)
+                return true
+            elseif masterHP1 then --cure                
+                pet:castSpell(level >= 80 and 6 or level >= 61 and 5 or level >= 41 and 4 or level >= 21 and 3 or level >= 11 and 2 or 1, master)
+                return true
+            end
+            local casted = buffPlayer(master,pet,level)
+            if casted == true then return true end
+            if partyHP then -- cure                
+                for _,member in ipairs(party) do
+                    if member:getHPP() < 50 then                    
+                        pet:castSpell(level >= 80 and 6 or level >= 61 and 5 or level >= 41 and 4 or level >= 21 and 3 or level >= 11 and 2 or 1, member)
+                        return true
+                    end
+                end      
+            end                       
+            for _,member in ipairs(party) do                
+                if math.random(0,99) < 50 then -- pick a player somewhat at random                    
+                    casted = buffPlayer(member,pet,level)
+                    if casted == true then return true end
+                end
+            end
+            if masterHP2 then -- low cure                
+                pet:castSpell(level >= 30 and 3 or level >= 20 and 2 or 1, master)
+                return true
+            end                       
+        end        
+    end
+    return false
+end;
+
+function buffPlayer(player,pet,level)        
+    if not player:hasStatusEffect(dsp.effect.PROTECT) and level >= 7 then -- protect        
+        local sp = level >= 76 and 47 or level >= 63 and 46 or level >= 47 and 45 or level >= 27 and 44 or 43                
+        pet:castSpell(sp, player)
+        return true
+    end
+    if not player:hasStatusEffect(dsp.effect.SHELL) and level >= 17 then -- shell        
+        pet:castSpell(level >= 76 and 52 or level >= 68 and 51 or level >= 57 and 50 or level >= 37 and 49 or 48, player)
+        return true
+    end
+    if not player:hasStatusEffect(dsp.effect.HASTE) and level >= 40 then -- haste        
+        pet:castSpell(57, player)
+        return true
+    end
+    if not player:hasStatusEffect(dsp.effect.REGEN) and level >= 21 then -- regen        
+        pet:castSpell(level >= 86 and 477 or level >= 66 and 111 or level >= 44 and 110 or 108, player)
+        return true
+    end
+    return false
+end;
+
+function getGearMod(master)
+    -- Summoner's spats
```

Comment:
Yeah there was no mod, wasn't totally familiar with how to create a new one, didn't want to introduce bugs so went this route, especially since there is only 1 single piece of gear (and its HQ counterpart) that has this stat boost.

Reply:
Pinging @helixhamin :wink: